lib/src/config_handler.dart

```diff
@@ -79,11 +96,25 @@ class ConfigHandler {
     disableTelemetryRegex.allMatches(configString).forEach((element) {
       // Conditional for recording telemetry as being disabled
       if (element.group(1) != ';' && element.group(2) == '0') {
-        telemetryEnabled = false;
+        _telemetryEnabled = false;
       }
     });
   }
 
+  /// Returns the telemetry state from the config file
+  ///
+  /// Method will reparse the config file if it detects that the
+  /// last modified datetime is different from what was parsed when
+  /// the class was initialized
+  bool get telemetryEnabled {
+    if (configFile.lastModifiedSync() != fileLastModified) {
+      parseConfig();
+      fileLastModified = configFile.lastModifiedSync();
+    }
+
+    return _telemetryEnabled;
+  }
+
```

Comment:
@christopherfujino this is the logic now, checks if the file was modified; if it was, it will reparse – if not, then we will just return the last state

Reply:
SGTM